reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -3184,23 +3184,25 @@ public final T blockLast(Duration timeout) {
 				catch (Exception e) {
 					return Mono.error(e);
 				}
-				if (v == null) {
-					return Mono.onAssembly(new MonoSupplier<>(listSupplier()));
-				}
-				return Mono.just(v).map(u -> {
+				return Mono.fromCallable(() -> {
```

Comment:
@simonbasle I believe a single `fromCallable` is better than `just` + `map` and easier to follow. WDYT?

Reply:
yeah that works for me